api/report.py

```diff
@@ -0,0 +1,63 @@
+from models.report import Report, ReportStatus
+
+from models.campaign import Campaign
+from flask_restx import Resource
+from core import Namespace
+
+
+api = Namespace('reports', description='Report in the system')
+
+report_model = api.model(Report)
+
+@api.route('/')
+class ReportListAPI(Resource):
+    '''
+    body: {
+        campaign_name: string
+        group: string
+        user: string
+    }
+    '''
+    def post(self):
+        campaign_name = api.payload['campaign_name']
+        group = api.payload['group']
+        campaign = Campaign.get_by_name(campaign_name)
+        new_report = Report()
+        new_report.status = ReportStatus.NOT_YET_DONE
+        new_report.campaign_name = campaign_name
+        new_report.content = ''
+        new_report.group = group
+        new_report.save()
+        campaign.reports.append(new_report)
+        campaign.save()
+        return new_report.dict()
+
+@api.route('/search/<string:campaign>/<string:group>/')
+@api.param('campaign', 'Campaign name')
+@api.param('group', 'Group path related to the report')
+class ReportSearchAPI(Resource):
+    @api.marshal_with(report_model)
+    def get(self, campaign, group):
+        result = Report.query({
+            'campaign_name': campaign,
+            'group': group,
+        })
+
+        if len(result) == 0:
```

Comment:
Just a general tip/advice - you usually don't need to check for `len` to be equal to zero, just do `if not result`. This will also check for `result` being `None`. Basically, anything that has length, can be checked without explicit `len`, it will first check if the variable is `None` and if it's not `None` it will implicitly check if length is 0, e.g. empty list/dict/set is `False`, empty string is also `False`.